kv/db.go

```diff
@@ -218,6 +69,28 @@ func (s *DBServer) RegisterRPC(rpcServer *rpc.Server) error {
 func (s *DBServer) executeCmd(argsI gogoproto.Message) (gogoproto.Message, error) {
 	args := argsI.(proto.Request)
 	reply := args.CreateReply()
+	if err := verifyRequest(args); err != nil {
+		return nil, err
+	}
 	s.sender.Send(context.TODO(), proto.Call{Args: args, Reply: reply})
 	return reply, nil
 }
+
+// verifyRequest checks for illegal inputs in request proto and
+// returns an error indicating which, if any, were found.
+func verifyRequest(args proto.Request) error {
+	switch t := args.(type) {
+	case *proto.EndTransactionRequest:
+		if t.InternalCommitTrigger != nil {
+			return util.Errorf("EndTransaction request from public KV API contains commit trigger: %+v", t.GetInternalCommitTrigger())
```

Comment:
Changed this to use `RegisterPublic`, but it doesn't appear to have any effect on semantics, including the authentication tests which still fail when I restore them.